index.bs

```diff
@@ -405,38 +400,66 @@ spec: webidl
     };
   </pre>
   <p>
-    The steps to <dfn>retrieve a permission storage entry</dfn> of a
-    <a>permission storage identifier</a> are as follows:
+    The steps to <dfn export>retrieve the permission storage</dfn> of a
+    {{PermissionName}} <var>name</var> are as follows:
   </p>
   <ol>
-    <li>If the <a>user agent</a> has a {{PermissionStorage}} associated
-    with the <a>permission storage identifier</a> in its permission store,
-    it MUST return the {{PermissionStorage}}.
+    <li>Let <var>settings</var> be the <a>current settings object</a>.</li>
+    <li>
+      Let <var>store</var> be some <a>permission store</a> associated with the
+      <a>same origin</a> as <var>settings</var>. This store SHOULD be chosen
+      consistently for a given <a>environment settings object</a>.
```

Comment:
The current PR gives UAs enough flexibility to do this right:
> The UA may initialize new <a>permission stores</a> as empty or as a copy of the mappings of any other <a>permission store</a> associated with the <a>same origin</a>.

and 

> Let <var>stores</var> be a UA-selected subset of the <a>permission stores</a> … Write a mapping … into each <a>permission store</a> in <var>stores</var>

The UA would initialize realm-store/"B" from origin-store/"B", and if "B" is updated globally from another realm, it would just select the subset as all realm-stores + the origin-store.

I can drop the requirement for a per-realm store and just replace the "SHOULD be chosen consistently" from my original text with MUST, which will probably simplify it a bit.